pkg/visor/api.go

```diff
@@ -303,6 +307,40 @@ func (v *Visor) Uptime() (float64, error) {
 	return time.Since(v.startedAt).Seconds(), nil
 }
 
+/*
+ */
+
+// SetPrivacy implements API.
+func (v *Visor) SetPrivacy(p skyenv.Privacy) error {
```

Comment:
Not sure if its a wise move to make the visor binary dependent on the existence of the `skywire-cli` command being existent and in $PATH. Why not write to the config yourself here?

Reply:
Also, it appears to me like we are calling `SetPrivacy` function inside the `skywire-cli config priv set` call via RPC. And then inside this function we are calling the cli function again. Are we sure this works.

Reply:
it is my strong conviction that the visor should be writing its own configs. The cli is tasked with config generation and updating.

the issue we have had previously is with having multiple implementations of the same thing, and a certain bifurcation or lack of feature parity existed in that.

with this, the config is only written by the cli, and everything else calls upon that existing functionality.

the visor is, in a practical sense, dependant on skywire-cli. We don't explicitly support omitting skywire-cli from the installation. I don't know how you could practically or reliably run skywire without updating the config on version changes, that sort of thing. You would need skywire-cli at some point and there is no reason to separate the binaries.

lets please assume it's in the path and not go down such rabbit holes. If the cli cannot be assumed to be present with the visor i will begin to question my sanity.

it's not like it breaks the visor if there is a command not found error there.

Reply:
./skywire-cli config priv set -o ./local/privacy.json not working even tho `skywire-cli` is installed.
![image](https://user-images.githubusercontent.com/38801159/193628822-7a15401f-8964-4539-a8f8-124ddf5ba480.png)

Like rudi mentioned we are just running it in a loop and are not actually writing to the privacy config
> Also, it appears to me like we are calling SetPrivacy function inside the skywire-cli config priv set call via RPC. And then inside this function we are calling the cli function again. Are we sure this works.

And I agree with this too
> Not sure if its a wise move to make the visor binary dependent on the existence of the skywire-cli command being existent and in $PATH. Why not write to the config yourself here?

The code to set the address in privacy should be in `SetPrivacy` apu which the cli calls anyway. So that cycle is cli -> SetPrivacy -> cli.

Reply:
The way the code is written right now is overly complicated and the API part of it won't even work because it simply calls itself. I think there is a way to fix it by simply moving the code that writes to the `privacy.json` to internal or to it's own pkg and that way both the instances will be using the exact same code.